tests/test_samples.py

```diff
@@ -1,170 +1,188 @@
 
-    target_line = "CertFile = "
-    replacement_line = target_line + "\"" + CERT_FILENAME + "\"\n"
-    overwrite_file_line(CONFIG_FILENAME, target_line, replacement_line)
+class StringDoesNotMatchRegEx(object):
+    def __init__(self, pattern):
+        self.pattern = pattern
 
-    target_line = "PrivateKey = "
-    replacement_line = target_line + "\"" + KEY_FILENAME + "\"\n"
-    overwrite_file_line(CONFIG_FILENAME, target_line, replacement_line)
+    def __eq__(self, other):
+        return not re.match(self.pattern, other)
 
-class SampleRunner(object):
 
-    def __init__(self, cmd, target=''):
-        self.cmd = cmd
-        self.target_file = target
-        self.process = None
-        self.output = "Not started"
+def run_sample(sample_file):
+    with open(sample_file) as f, \
+            patch.object(builtins, 'print') as mock_print:
+        sample_globals = {"__file__": sample_file}
+        exec(f.read(), sample_globals)  # pylint: disable=exec-used
+    return mock_print
 
 
-    def run(self, timeout):
-        def target():
-            self.process = subprocess.Popen(
-                [self.cmd, self.target_file],
-                stdout=subprocess.PIPE,
-                #stderr=subprocess.PIPE,
-            )
-            self.output = self.process.communicate()[0]
+class TempSampleFile(object):
 
-        thread = threading.Thread(target=target)
-        thread.start()
+    @property
+    def temp_file(self):
+        return self._temp_file
 
-        thread.join(timeout)
-        if thread.is_alive():
-            self.process.terminate()
-            thread.join()
+    def __init__(self, sample_filename):
+        self._temp_file = NamedTemporaryFile(mode="w+", delete=False)
+        self._temp_file.close()
+        os.chmod(self._temp_file.name, 0o777)
+        self.base_filename = sample_filename
+        self.write_file_line()
 
-        return self.output.decode('utf-8')
+    def write_file_line(self, target=None, replacement=None):
+        base_file = open(self.base_filename, 'r')
+        with open(self._temp_file.name, 'w+') as new_sample_file:
 
-class TestCoreHelpSample(BaseEpoClientTest):
+            for line in base_file:
+                if target != None and replacement != None:
+                    if line.startswith(target):
+                        line = replacement
+                new_sample_file.write(line)
 
-    def test_corehelp_example(self):
-        # Modify common/config files to work with local ".\test" directory
-        overwrite_common_py()
-        overwrite_config_cert_locations()
+            base_file.close()
+            new_sample_file.close()
+
+    def __del__(self):
+        self.temp_file.close()
+        os.remove(self._temp_file.name)
+
+
+class TestSamples(BaseClientTest):
+
+    SAMPLE_FOLDER = str(os.path.dirname(
+        os.path.dirname(
+            os.path.abspath(__file__)
+        )
+    ).replace("\\", "/")) + "/sample"
```

Comment:
To avoid some of the path delimiter replacement/concatenation code, you could use `os.path.join()` instead.